### PR TITLE
Added/Reworked unit tests for uno.go.

### PR DESCRIPTION
**Pull request by Jaredcscott**
I attempted to implement the MockDB in these unit tests.

### CODE REVIEW

File: server/uno_test.go [hunk 1]
```diff
@@ -5,11 +5,15 @@ import (
...
+	"os"
+	"errors"
 )
 
+
+
 // This function is meant to get a game and a player into the data base in a usable state for testing.
 func setupGameWithPlayer(database *db.DB) (*model.Game, *model.Player) {
-
+	os.Setenv("DB_TYPE", "MOCK")
```
Comment (logan-hendricks) [inline server/uno_test.go:16]: I really am confused... as this shouldn't be necessary. I'd love to review this after class if you have time.

**Comment by logan-hendricks** [comment]
@Jaredcscott On the right hand side of this page. You can Link which issues this PR belongs to:
image

Can you select the issues that should be connected with this PR?

**Comment by Jaredcscott** [comment]
Sure!

File: server/uno_test.go [hunk 2]
```diff
@@ -187,28 +193,32 @@ func TestCheckForCardInHand(t *testing.T){
...
 	// Lookup the player in the database to see if it is there
 	databasePlayer, err := database.LookupPlayer(player.ID)
 	assert.Nil(t, err, "could not find player")
 	// Test to see if the database player and the created player are the same
-	assert.Equal(t, player, databasePlayer)
+	assert.Equal(t, player, databasePlayer) 
 }
 
 func TestJoinGame(t *testing.T){
+	os.Setenv("DB_TYPE", "MOCK")
```
Comment (jak103) [inline server/uno_test.go:211]: Does this have to be done in each test or should this instead be done from the command line test call?

**Comment by jak103** [comment]
There is a failing test after recent updates to dev. Please fix the failing test and I'll get this merged.

File: server/uno_test.go (hunk 2 above)
Comment (Jaredcscott) [reply server/uno_test.go:211]: Hmm sounds like a better way, I am not sure how to actually go about doing that, do I set the environment variable in the same command as go test -cover? Or would I do an export DB_TYPE="MOCK" prior to running my tests?

File: server/uno_test.go [hunk 3]
```diff
@@ -120,11 +125,13 @@ func TestDrawCard(t *testing.T) {
 	assert.Equal(t, "It is not your turn to play", err.Error())
 	assert.Equal(t, 0, len(player2.Cards))
 	assert.Equal(t, 107, len(game.DrawPile))
-
+	*/
+	// TODO: This test should be rewritten to use a MockDB
```
[the remark this answers is not in the document]
Comment (Jaredcscott) [reply server/uno_test.go]: Okay I believe have fixed this.

File: server/uno_test.go (hunk 1 above)
Comment (Jaredcscott) [reply server/uno_test.go]: Haha I'm the one who was confused, there's a bit of new content here for me, so I had a little learning curve, but I think I understand now. Thanks for the help, here and in thew discord!

**Pull request merged**